FirebaseRemoteConfig/Sources/RCNConfigDBManager.m

```diff
@@ -861,12 +909,76 @@ - (void)loadExperimentWithCompletionHandler:(RCNDBCompletion)handler {
 
   NSArray *params = @[ key ];
   [self bindStringsToStatement:statement stringArray:params];
-  NSData *experimentData;
+  NSMutableArray *results = [self loadValuesFromStatement:statement];
+  return results;
+}
+
+- (void)loadRolloutWithCompletionHandler:(RCNDBCompletion)handler {
+  __weak RCNConfigDBManager *weakSelf = self;
+  dispatch_async(_databaseOperationQueue, ^{
+    RCNConfigDBManager *strongSelf = weakSelf;
+    if (!strongSelf) {
+      dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
General question: any reason we dispatch success result to `main queue` but failed one to global queue?

Reply:
I made a mistake here. For personalization, both success and failure are dispatched to global queue. For experiments, the failure is ignored and success dispatched to main. For rollout, I think we probably should follow the pattern as experiments?

Reply:
Following experiments makes sense to me since that more analogous to rollouts than p13ns. I'll take a closer look after submitting the review

Reply:
I would personally try get as much tasks off [main queue](https://developer.apple.com/documentation/dispatch/1452921-dispatch_get_main_queue?language=objc) as possible since it associates with the application’s main thread which is also the UI thread. Some reference to explain the difference between main thread and main queue: https://www.hackingwithswift.com/quick-start/concurrency/main-thread-and-main-queue-whats-the-difference

Reply:
Rollout metadata loading combined into `loadMetadataWithBundleIdentifier`, which is dispatched to global queue, now rollout loading is also directed to global queue as a result..